### PR TITLE
New setting: log_step_size

### PR DESCRIPTION
**Pull request by psinger**
Introduces a new setting `log_step_size`.

Specifies the interval for logging during training. Two options are available:

- **Absolute**: The default setting. Uses the total number of training samples processed as the x-axis for logging.
- **Relative**: Uses the proportion of training data seen so far as the x-axis for logging.

The new relative setting will always range from 0 to 1, even in case of multiple epochs. But it solves the proposed change in https://github.com/h2oai/h2o-llmstudio/issues/700 meaning that multiple runs with different amounts of data can be compared relatively.

Additionally, cleaned some `step_size` specifications for internal logging. Please confirm when review if I didnt miss anything there for that to be required.

image

### CODE REVIEW

**Review by pascal-pfeiffer — APPROVED** [review]
Very handy! All lgtm, thank you.

**Comment by pascal-pfeiffer** [comment]
And let's add the tooltip to the documentatio page, too, please.
